### ISSUE TITLE
pytype of a larger but not huge standalone script xen-bugtool never finishes

### ISSUE THREAD

**Issue by bernhardkaindl** [google/pytype#1577]
Running pytype on this script on Ubuntu 22.04 never finishes (Interrupted it after 42 minutes):

https://github.com/xenserver/status-report/blob/master/xen-bugtool
 
It is standalone and needs uses only one optional non-stdlib module that does not need to be present.

**Comment by rchen152** [comment]
Yeah, 42 minutes (!) definitely seems suspiciously long. Let me check if pytype is entering an infinite loop somewhere.
